src/librustc_trans/mir/statement.rs

```diff
@@ -57,6 +58,22 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
                     self.trans_rvalue(bcx, tr_dest, rvalue, debug_loc)
                 }
             }
+            mir::StatementKind::StorageLive(ref lvalue) => {
+                if let Some(index) = self.mir.local_index(lvalue) {
+                    if let LocalRef::Lvalue(tr_lval) = self.locals[index] {
+                        base::Lifetime::Start.call(&bcx, tr_lval.llval);
+                    }
+                }
+                bcx
+            }
+            mir::StatementKind::StorageDead(ref lvalue) => {
+                if let Some(index) = self.mir.local_index(lvalue) {
+                    if let LocalRef::Lvalue(tr_lval) = self.locals[index] {
+                        base::Lifetime::End.call(&bcx, tr_lval.llval);
+                    }
+                }
+                bcx
+            }
```

Comment:
We need to do nothing if the local is not an `Lvalue`.
Using a method taking `lvalue` and the `base::Lifetime` would work, yeah.